### DISCUSSION TITLE
How to obtain statistical data and file information

### DISCUSSION THREAD

**Discussion by antwal** [sn4k3/UVtools#555] [Q&A] [answered]
It would be possible to extract separately or save an additional file in text or json format with some information contained in the file such as:

Layer Height
Resolution X / Y
Total Height Millimeter
Print Time
Used Material
Material Cost
Machine
AntiAliasLevel
Material name (if available)

Currently I copy this data manually;

if available a command line function would also be fine, but I couldn't find anything in the documentation

**Comment by sn4k3** [comment, marked as answer]
There are multiple ways to obtain that:

1. `File - Extract`, it will dump everything inc. layers which may be overkill
2. `Layer actions - Export to HTML`, will create a information file with all settings, you can exclude layers and other optional content
3. `File - Terminal` and create a snippet that you can copy paste to generate that info and either dump or set to clipboard and/or file
4. A [powershell](https://github.com/sn4k3/UVtools/tree/master/Scripts/UVtools.Powershell) or [python](https://github.com/sn4k3/UVtools/tree/master/Scripts/UVtools.Python) script where you can just input for file and generate your content as you need
5. Command line, see `UVtoolsCmd print-properties`:

```
Description:
  Prints available properties

Usage:
  UVtoolsCmd print-properties [<input-file>] [options]

Arguments:
  <input-file>  Input file to open and read

Options:
  -n, --names <names>  Prints only the name matching properties
  -b, --base           Prints only the base properties of the file
  --partial-mode       Fast load the file in partial mode
  -?, -h, --help       Show help and usage information
```

Example: `UVtoolsCmd print-properties UVtools_demo_file.sl1 --partial-mode -b -n:LayerHeight -n:ResolutionX -n:ResolutionY -n:PrintHeight -n:PrintTime -n:MaterialMilliliters -n:MaterialCost -n:MachineName -n:AntiAliasing -n:MaterialName`

```bash
Opening file UVtools_demo_file.sl1: Done in 0.12s
Listing properties:
----------------------
ResolutionX: 1440
ResolutionY: 2560
AntiAliasing: 4
LayerHeight: 0.1
PrintTime: 491.5
MaterialMilliliters: 0
MaterialCost: 0
MaterialName: BlueCast EcoGray @0.1
MachineName: EPAX X1
PrintHeight: 1
----------------------
Total properties: 10
```

To save that content to a file, repeat same command but append: ` > to_my_file.txt`
To parse contents via script you can use [regex](https://regex101.com/r/SdVbqU/1): `^(\w+): (.+)$`

**Reply by antwal** [reply]
command line is fine, but is the PrintTime in seconds?

**Reply by sn4k3** [reply]
It is

**Reply by antwal** [reply]
perfect

**Reply by sn4k3** [reply]
You also have: [PrintTimeHours](https://github.com/sn4k3/UVtools/blob/4ff621a080e28e971180b91789acb8ae6ce4937f/UVtools.Core/FileFormats/FileFormat.cs#L2800) and [PrintTimeString](https://github.com/sn4k3/UVtools/blob/4ff621a080e28e971180b91789acb8ae6ce4937f/UVtools.Core/FileFormats/FileFormat.cs#L2805)

**Reply by antwal** [reply]
hi, on most of the files when I use "MaterialMilliliters" it always returns the value 0, instead "MaterialCost" is always calculated, how can I do to recover the value of the quantity of the material from all the files?Thanks

**Reply by sn4k3** [reply]
For file formats that don't have "MaterialMilliliters" information inbuit you need let UVtools to compute that, for that you need to open files in full mode, in other words, you need to remove the `--partial-mode` flag from command

**Reply by antwal** [reply]
ok
